### PR TITLE
Jetpack-connect: Redirect to jetpack auth endpoint

### PR DESCRIPTION
**Pull request by johnHackworth**
This PR modifies the dummy jetpack connect url input box to really redirect to the self-host server endpoint that would start the remote auth request.

Depends on https://github.com/Automattic/wp-calypso/pull/3854 being merged to work

image

This is how it works for a user that is logged in their own server:
https://cloudup.com/cvcLj7bgZ0T

And this how it looks when they need to log there before continuing:
https://cloudup.com/cga6KSG4hw0

Testing:
========
1. You need your site to be running this branch of jetpack: https://github.com/Automattic/jetpack/pull/3364
2. Go to http://calypso.dev:3000/jetpack/connect, enter the url of your site, and ~watch the magic happen :) ~ Not yet! We need to merge https://github.com/Automattic/wp-calypso/pull/2974 to make this work as in the videos ... right now this would redirect you to your wp-admin, which would redirect you to an url that calypso doesn't know how to handle... so it would take you to the theme selector, for some reason. But the main purpose of this PR was to add that calypso => wp-admin redirection, so it works :D

ping @roccotripaldi @oskosk @alternatekev

### CODE REVIEW

File: client/signup/jetpack-connect/index.jsx [hunk 1]
```diff
@@ -31,31 +40,38 @@ export default React.createClass( {
...
-		window.location = this.refs.siteUrlInputRef.state.value + authURL;
+		window.location = this.getCurrentUrl() + authURL;
+	},
+
+	showNotExistsError() {
+		this.setState( { siteStatus: 'notExist' } );
+	},
+
+	checkIfUrlIsReadyForJetpack( url ) {
+		var promise = new Promise( function( resolve, reject ) {
```
Comment (lezama) [inline client/signup/jetpack-connect/index.jsx]: `const` ?

Comment (johnHackworth) [inline client/signup/jetpack-connect/index.jsx]: :+1: !

Comment (lezama) [inline client/signup/jetpack-connect/index.jsx]: or `return` :)

**Comment by oskosk** [comment]
LGTM !

**Pull request merged**